cime_config/machines/cmake_macros/universal.cmake

```diff
@@ -1 +1,11 @@
 set(SUPPORTS_CXX "FALSE")
+
+# Default guess at KOKKOS_OPTIONS. These can be overridden by
+# appending -D${OPTION_NAME}=Off. The CMAKE_CXX_COMPILER will
+# be handled automatically by CIME unless explicitly set in
+# KOKKOS_OPTIONS. CMAKE_INSTALL_PREFIX will be handled automatically
+# by CIME always.
+set(KOKKOS_OPTIONS "-DKokkos_ENABLE_SERIAL=On")
+if (compile_threaded)
+  string(APPEND KOKKOS_OPTIONS " -DKokkos_ENABLE_OPENMP=On")
+endif()
```

Comment:
I think `-DKokkos_ENABLE_OPENMP=On` had always been omitted (default is `off`) even with `compile_threaded=true`, because all available CPU cores are typically occupied with OpenMP threads outside of kokkos. With this option `On`, there are build issues with pgi and ibm on Ascent for all (11) threaded tests
- https://my.cdash.org/viewTest.php?onlydelta&buildid=2240179
- https://my.cdash.org/viewTest.php?onlydelta&buildid=2242495

Can we revert to the previous configuration? Unless there's a use-case for nested threading (in addition to `Kokkos_ENABLE_OPENMPTARGET`)

Reply:
I don't disagree, but I'll just add some observations as long as I took a look:
1. The build error is in kokkos.bldlog. e3sm.bldlog hasn't been created yet.
2. The error is in a Kokkos file: `#error directive: "You enabled Kokkos OpenMP support without enabling OpenMP in the compiler!"`
3. Thus, likely the issue can be resolved simply by including the equivalent of `-fopenmp` for the PGI compiler.
4. The fact that Kokkos isn't doing this automatically in its build system might indicate it's not being given the right information about the compiler, but that's just speculation.
5. @amametjanov is right that Kokkos threads aren't used in E3SMv2. But the only Kokkos component in v2, SL transport, knows not to use the OpenMP execution space, so it's harmless to enable the OpenMP execution space in the build.
6. I believe that Jim made this change to support SCREAM.